aws/resource_aws_ssm_patch_baseline.go

```diff
@@ -219,6 +224,11 @@ func resourceAwsSsmPatchBaselineUpdate(d *schema.ResourceData, meta interface{})
 		params.GlobalFilters = expandAwsSsmPatchFilterGroup(d)
 	}
 
+	if d.HasChange("tags") {
+		if err := setTagsSSM(ssmconn, d, d.Id(), ssm.ResourceTypeForTaggingPatchBaseline); err != nil {
+			return fmt.Errorf("error setting SSM Patch Baseline tags: %s", err)
+		}
+	}
```

Comment:
Similar to above - I'd suggest moving this logic below `UpdatePatchBaseline` & related error handling, so we can retain the graceful handling of 404.